src/django/pfb_analysis/models.py

```diff
@@ -101,6 +103,7 @@ def set_boundary_file(self, geom):
                         zip_handle.write(os.path.join(tmpdir, shpfile), shpfile)
             boundary_file = File(open(zip_filename))
             self.boundary_file = boundary_file
+            self.geom = geom
```

Comment:
Shortcut by directly setting geom here, so that we don't create a shapefile from the geom to upload, then in save recreate the shapefile locally again just so we can get the geom back from it that we already have.